src/passes/Outlining.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This pass is a placeholder, encapsulating a suffix tree data structure that
+// will eventually be moved to src/support
+
+#include "ir/module-utils.h"
+#include "ir/names.h"
+#include "pass.h"
+#include "src/ir/utils.h"
+#include "stringify-walker.h"
+#include "wasm-builder.h"
+#include "wasm-stack.h"
+#include <wasm.h>
+
+namespace wasm {
+
+void StringifyWalker::walkModule(Module* module) {
+  this->pushTask(StringifyWalker::handler, nullptr);
+  ModuleUtils::iterDefinedFunctions(*module, [&](Function* func) {
+    this->emitFunctionBegin(this);
+    this->walk(func->body);
+  });
+}
+
+void StringifyWalker::scanChildren(StringifyWalker* stringify,
+                                   Expression** currp) {
+  Expression* curr = *currp;
+  switch (curr->_id) {
+    case Expression::Id::BlockId: {
+      auto* block = curr->dynCast<Block>();
+      auto blockIterator = block->list.end();
+      while (blockIterator != block->list.begin()) {
+        blockIterator--;
+        auto& child = block->list[blockIterator.index];
+        [[maybe_unused]] auto name = getExpressionName(child);
+        // std::cout << "Pushing a task to call StringifyWalker::scan with block
+        // list item: " << name  << " " << child << std::endl;
+        stringify->pushTask(StringifyWalker::scan, &child);
+      }
+      break;
+    }
+    case Expression::Id::IfId: {
+      auto* iff = curr->dynCast<If>();
+      stringify->pushTask(StringifyWalker::scan, &iff->ifFalse);
+      // std::cout << "Pushing an task to call StingifyWalker::scan on ifFalse "
+      // << std::endl;
+      stringify->pushTask(StringifyWalker::scan, &iff->ifTrue);
+      // std::cout << "Pushing an task to call StingifyWalker::scan on ifTrue "
+      // << std::endl;
+      stringify->pushTask(StringifyWalker::scan, &iff->condition);
```

Comment:
Since the condition is not actually inside the `if` structure in the binary, it should be pushed during the normal `scan` rather than here in `scanChildren`.